lib/msprime.c

```diff
@@ -3746,6 +4448,64 @@ msp_apply_fixed_events(msp_t *self, double time)
     return ret;
 }
 
+static int MSP_WARN_UNUSED
+msp_hulls_init_counts(msp_t *self, population_id_t pop, label_id_t label)
+{
+    int ret = 0;
+    avl_tree_t *hulls_left, *hulls_right;
+    avl_node_t *node_left, *node_right, *search;
+    hullend_t hullend_target;
+    hullend_t *hullend, *floor_ptr;
+    hull_t *hull;
+    fenwick_t *coal_mass;
+    uint64_t visited, num_ending_before_hull, value;
+    double left;
+
+    visited = 0;
+    hulls_left = &self->populations[pop].hulls_left[label];
+    hulls_right = &self->populations[pop].hulls_right[label];
+    coal_mass = &self->populations[pop].coal_mass_index[label];
+
+    /* iterate over avl nodes */
+    for (node_left = hulls_left->head; node_left != NULL; node_left = node_left->next) {
+        tsk_bug_assert(avl_count(hulls_left) > 0);
+        num_ending_before_hull = 0;
+        hull = (hull_t *) node_left->item;
+        left = hull->left;
+        hullend_target.position = left;
+        hullend_target.insertion_order = UINT64_MAX;
+        avl_search_closest(hulls_right, &hullend_target, &search);
+        if (search != NULL) {
+            floor_ptr = (hullend_t *) search->item;
+            if (floor_ptr->position > left) {
+                search = search->prev;
+            }
+            if (search != NULL) {
+                num_ending_before_hull = (uint64_t) avl_index(search) + 1;
```

Comment:
This isn't being hit in C tests

Reply:
note to self: this requires setting up an initial state where the samples do not all span entire sequence length.